emulator.js

```diff
@@ -80,122 +79,138 @@ function (Cpu6502, Video, SoundChip, models, DdNoise, Cmos,  utils,fdc,tokeniser
       }
       frame++;});
 
-        // Set up our BBC Micro emulator
+      // Set up our BBC Micro emulator default
+      if (tweetdisk==null){
+
         processor = new Cpu6502(model, dbgr, video, soundChip, new DdNoise.FakeDdNoise(), new Cmos());
         await processor.initialise();
-        await runUntilInput();
-
-        // Use the tokeniser if the program contains non-ASCII characters or if
-        // it doesn't have any line numbers.
-        if (/[^\0-\x7e]/.test(input) || !/^ *[0-9]/m.test(input)) {
-          /* Tokeniser input method */
-          var t         = await tokeniser.create();
-          var tokenised = await t.tokenise(input);
-
-          var page = processor.readmem(0x18) << 8;
-          for (var i = 0; i < tokenised.length; ++i) {
-            processor.writemem(page + i, tokenised.charCodeAt(i));
-          }
-          var end = page + tokenised.length;
-          var endLow = end & 0xff;
-          var endHigh = (end >>> 8) & 0xff;
-          // Set LOMEM.
-          processor.writemem(0x00, endLow);
-          processor.writemem(0x01, endHigh);
-          // Set VARTOP.
-          processor.writemem(0x02, endLow);
-          processor.writemem(0x03, endHigh);
-          // Set TOP.
-          processor.writemem(0x12, endLow);
-          processor.writemem(0x13, endHigh);
-
-          input="RUN\r";
-        } else {
-          input=input.replace(/[\n]/g,'\r');
-          if (!input.endsWith('\r')) {
-            input=input+'\r';
-          }
-          if (!/\r\s*RUN\s*\r/.test(input)) {
-            input=input+"RUN\r";
-          }
-        }
+            await runUntilInput();
+        var runcmd = "RUN\r";
+      }
+      else {
+        // Tweetdisk
+        processor = new Cpu6502(model, dbgr, video, soundChip, new DdNoise.FakeDdNoise(), new Cmos());
+        await processor.initialise();
+        await processor.fdc.loadDisc(0, fdc.discFor(processor.fdc, "tweetdisk", tweetdisk.data));
+        processor.sysvia.keyDown(16); // shift boot
+          await runFor(2000000);
+        processor.sysvia.keyUp(16); //
+        var runcmd = "";
```

Comment:
This `runcmd` mechanism seems superfluous - it's only used in the tokenisation code which is only executed if `tweetdisk == null`.